### ISSUE TITLE
[Direct upload] Possibilité de configuration / paramétrage sur le modèle d'une pièce justificative.

### ISSUE THREAD

**Issue by SylvainBlancheDNUM** [betagouv/demarches-simplifiees.fr#6995]
**Problème rencontré**

Pour les champs de type "file", l'activation du mécanisme "direct upload" est dépendante d'un paramètre "direct_upload" qu'on peut trouver dans différents fichiers .haml ou aussi dans app/helpers/attachment_upload_helper.rb .

A la création d'une démarche il semble que pour l'upload du modèle d'une pièce justificative, on ne puisse pas "jouer" sur un paramètre de ce type qui nous permettrait de désactiver facilement ce mécanisme.

**Questions**

Est ce bien le cas ou il y a un autre moyen pour ce champ ?

Est qu'une évolution en ce sens est possible ?

**Comment by tchak** [comment]
Effectivement, ce n’est pas possible de désactiver "direct upload" pour les modèles d'attestation des champs. La raison est que l'éditeur des champs est un composant dynamique qui communique avec le backend via une API interne en JSON et donc il n'est pas possible de passer un fichier dans le corps de l'appel d'API. Si on voulait ajouter la possibilité de désactiver le “direct upload" dans ce contexte, il faudrait implémenter un nouveau endpoint sur cette API interne pour l'upload de fichiers et implémenter une logique de “garbage collection" si le fichier n'est pas attaché. En gros, ça revient à re-implanter un “direct upload du pauvre”. Étant donné que c'est un changement non trivial et que nous n'avons pas l'usage sur l'instance principale de DS ça va être difficile pour nous de dédier des ressources à un changement dans ce sens.

Nous avons pour projet de retravailler sur l’éditeur des champs dans les mois à venir (lié au projet du support des champs conditionnels). Peut être qu’à ce moment on pourrait réfléchir à la question.

**Comment by SylvainBlancheDNUM** [comment]
Ok, merci pour les détails.
